src/broadcast.jl

```diff
@@ -0,0 +1,29 @@
+Base.Broadcast._containertype(::Type{<:DataValue}) = DataValue
+
+Base.Broadcast.promote_containertype(::Type{Any}, ::Type{DataValue}) = DataValue
+Base.Broadcast.promote_containertype(::Type{DataValue}, ::Type{Any}) = DataValue
+Base.Broadcast.promote_containertype(::Type{Tuple}, ::Type{DataValue}) = Tuple
+Base.Broadcast.promote_containertype(::Type{DataValue}, ::Type{Tuple}) = Tuple
+
+Base.Broadcast.broadcast_indices(::Type{DataValue}, A) = ()
+
+Base.Broadcast._unsafe_get_eltype(x::DataValue) = eltype(x)
+
+Base.Broadcast._broadcast_getindex_eltype(::Type{DataValue}, T::Type) = Type{T}
+Base.Broadcast._broadcast_getindex_eltype(::Type{DataValue}, A) = typeof(A)
+
+Base.@propagate_inbounds Base.Broadcast._broadcast_getindex(::Type{DataValue}, A, I) = A
+
+@inline function Base.Broadcast.broadcast_c(f, ::Type{DataValue}, a...)
+    nonnull = all(Base.hasvalue, a)
```

Comment:
Since this shares `Base.hasvalue` and `Base.unsafe_get` with `Nullable`, when mixing `DataValue` with `Nullable` the result is null if any argument is null. That does make some degree of sense to me; but if so, what's the difference between `DataValue` and `Nullable` if they're interchangeable in this way?